src/custom/hooks/useMetadataDoc.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react'
+import { useActiveWeb3React } from 'hooks/web3'
+import { generateReferralMetadataDoc, AppDataDoc } from 'utils/metadata'
+
+export function useMetadataDoc() {
+  const { account } = useActiveWeb3React()
+  const [document, setDocument] = useState<AppDataDoc>(generateReferralMetadataDoc(account || ''))
+
+  useEffect(() => {
+    const fetchDocument = async (): Promise<AppDataDoc | undefined> => {
```

Comment:
I believe the fetch call is part of #961. From what I understood, this ticket should only handle parsing the referral query param from the url and return either the default metadata object or the metadata object with the referrer, so it's a matter of integrating the `useParseReferralQueryParam` hook with the functions in the `metadata.ts` file.

@alfetopito could you clarify this?

Reply:
Yes, there's no fetching in the scope of this task.
Ramiro's comment is correct.